drivers/wifi/wfx200/wfx200.c

```diff
@@ -0,0 +1,499 @@
+/* SiLabs WFX200 WiFi Module with SPI
+ * or SDIO Interface
+ *
+ * Copyright (c) 2022 grandcentrix GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+
+#define LOG_MODULE_NAME wifi_wfx200
+#define LOG_LEVEL CONFIG_WIFI_LOG_LEVEL
+
+#include <sl_wfx.h>
+#undef BIT
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(LOG_MODULE_NAME);
+
+#include <string.h>
+#include <errno.h>
+
+#include <zephyr.h>
+#include <device.h>
+#include <drivers/gpio.h>
+#include <drivers/spi.h>
+#include <net/net_pkt.h>
+#include <net/net_if.h>
+#include <net/ethernet.h>
+
+#include "wfx200_internal.h"
+
+static void wfx200_iface_init(struct net_if *iface)
+{
+	const struct device *dev = net_if_get_device(iface);
+	struct wfx200_data *context = dev->data;
+
+	LOG_DBG("Interface init");
+	context->iface = iface;
+	net_if_set_link_addr(iface, context->sl_context.mac_addr_0.octet,
+			     sizeof(context->sl_context.mac_addr_0.octet),
+			     NET_LINK_ETHERNET);
+	ethernet_init(iface);
+	context->state = WFX200_STATE_INTERFACE_INITIALIZED;
+	net_if_flag_set(iface, NET_IF_NO_AUTO_START);
+}
+
+int wfx200_send(const struct device *dev, struct net_pkt *pkt)
+{
+	struct wfx200_data *context = dev->data;
+	size_t len = net_pkt_get_len(pkt), frame_len;
+	sl_wfx_send_frame_req_t *tx_buffer;
+	sl_status_t result;
+	sl_wfx_interface_t interface;
+	int status = 0;
+
+	switch (context->state) {
+	case WFX200_STATE_AP_MODE:
+		if (!(context->sl_context.state & SL_WFX_AP_INTERFACE_UP)) {
+			status = -EIO;
+			goto out;
+		}
+		interface = SL_WFX_SOFTAP_INTERFACE;
+		break;
+	case WFX200_STATE_STA_MODE:
+		if (!(context->sl_context.state & SL_WFX_STA_INTERFACE_CONNECTED)) {
+			status = -EIO;
+			goto out;
+		}
+		interface = SL_WFX_STA_INTERFACE;
+		break;
+	default:
+		status = -ENODEV;
+		goto out;
+	}
+	frame_len = SL_WFX_ROUND_UP(len, 2);
+
+	result = sl_wfx_allocate_command_buffer((sl_wfx_generic_message_t **)&tx_buffer,
+						SL_WFX_SEND_FRAME_REQ_ID,
+						SL_WFX_TX_FRAME_BUFFER,
+						frame_len + sizeof(sl_wfx_send_frame_req_t));
+	if (result != SL_STATUS_OK) {
+		status = -ENOMEM;
+		goto out;
+	}
+
+	if (net_pkt_read(pkt, tx_buffer->body.packet_data, len)) {
+		status = -EIO;
+		goto out_free;
+	}
+
+	result = sl_wfx_send_ethernet_frame(tx_buffer,
```

Comment:
First of all thanks for an excellent driver update! 
Looks pretty good, but we have a just couple of questions. Here is the first one. 

[FreeRTOS drivers for WFX](https://github.com/SiliconLabs/wfx-fullMAC-tools/blob/wifi_examples_ssv4/Examples/SiliconLabs/commissioning/freeRTOS/SLSTK3701A/src/sl_wfx_host/sl_wfx_task.c#L109) use dedicated thread for sending data, and, most importantly, queue. 


Is it not necessary in Zephyr because queuing is done in higher levels of stack?
Namely, in [net_if_queue_tx](https://github.com/rvoell-gcx/zephyr/blob/wfx200/subsys/net/ip/net_if.c#L335) ?